packages/cubejs-backend-cloud/src/deploy.ts

```diff
@@ -61,3 +62,53 @@ export class DeployDirectory {
     });
   }
 }
+
+type DeployHooks = {
+  onStart?: Function,
+  onUpdate?: Function,
+  onUpload?: Function,
+  onFinally?: Function
+};
+export class DeployController {
+  public constructor(
+    protected readonly cubeCloudClient: CubeCloudClient,
+    protected hooks: DeployHooks = {}
```

Comment:
`hooks` should be `readonly`, because you don't change the reference to the object.